### PR TITLE
Use `odo service list -o json --context <compName>' to get list of provisioned service instances

### PR DESCRIPTION
**Pull request by sudhirverma**
Fix: #726

### CODE REVIEW

File: src/odo.ts [hunk 1]
```diff
@@ -472,10 +473,20 @@ class OdoModel {
         return this.pathToObject.get(path);
     }
 
+    public setStringToContext(name: string, uri: Uri) {
```
Comment (mohitsuman) [inline src/odo.ts]: @sudhirverma What's the significance of this function?

Comment (sudhirverma) [reply src/odo.ts]: I am storing the context path for application to get list of services.

Comment (mohitsuman) [reply src/odo.ts]: For an application, there can be multiple components and services, and hence multiple context paths. How are we handling that scenario ?

Comment (sudhirverma) [reply src/odo.ts]: for an application, I am storing only one context path and updating the context path base on the newly created component and from that context path we can get all the services for that application.

File: src/odo.ts [hunk 2]
```diff
@@ -389,7 +389,7 @@ export interface Odo {
     getClusters(): Promise<OpenShiftObject[]>;
     getProjects(): Promise<OpenShiftObject[]>;
     loadWorkspaceComponents(event: WorkspaceFoldersChangeEvent): void;
-    addWorkspaceComponent(WorkspaceFolder: WorkspaceFolder, component: OpenShiftObject);
+    addWorkspaceComponent(WorkspaceFolder: WorkspaceFolder, component: OpenShiftObject): void;
```
Comment (dgolovin) [inline src/odo.ts]: If you want to improve, clean up, fix issues like this, you are welcome to send additional PR. This PR should have changes to compiler or linter settings to report issues like this during the build.
Please remove it form this PR.

File: src/odo.ts [hunk 3]
```diff
@@ -101,14 +101,14 @@ export class Command {
...
+    static listStorageNames() {
         return `odo storage list -o json`;
     }
     static printOcVersion() {
         return 'oc version';
     }
-    static listServiceInstances(project: string, app: string) {
-        return `oc get ServiceInstance -o jsonpath="{range .items[?(.metadata.labels.app\\.kubernetes\\.io/part-of == \\"${app}\\")]}{.metadata.labels.app\\.kubernetes\\.io/instance}{\\"\\n\\"}{end}" --namespace ${project}`;
+    static listServiceInstances(context: string) {
+        return `odo service list -o json --context ${context}`;
```
Comment (dgolovin) [inline src/odo.ts]: Restore previous parameters for the method and use them with --project and --app options. Do not forget to remove --context option.

File: src/odo.ts [hunk 4]
```diff
@@ -101,14 +101,14 @@ export class Command {
     static listCatalogServices () {
         return `odo catalog list services -o json`;
     }
-    static listStorageNames(project: string, app: string, component: string) {
```
Comment (dgolovin) [inline src/odo.ts]: It should be part of different PR. Please remove form this one.

File: src/odo.ts [hunk 5]
```diff
@@ -443,6 +443,7 @@ class OdoModel {
     private pathToObject = new Map<string, OpenShiftObject>();
     private contextToObject = new Map<Uri, OpenShiftObject>();
     private contextToSettings = new Map<Uri, ComponentSettings>();
+    private stringToContext = new Map<string, Uri>();
```
Comment (dgolovin) [inline src/odo.ts]: Not needed.

File: src/odo.ts (hunk 1 above)
Comment (dgolovin) [reply src/odo.ts]: This is not needed.

File: src/odo.ts [hunk 6]
```diff
@@ -642,7 +649,8 @@ export class OdoImpl implements Odo {
         let apps: string[] = this.loadItems(result).map((value) => value.metadata.name);
         apps = [...new Set(apps)]; // remove duplicates form array
         // extract apps from local not yet deployed components
-        OdoImpl.data.getSettings().forEach((component) => {
```
Comment (dgolovin) [inline src/odo.ts]: Please revert these changes.

File: src/odo.ts [hunk 7]
```diff
@@ -689,9 +697,10 @@ export class OdoImpl implements Odo {
         });
         const targetAppName = application.getName(),
             targetPrjName = application.getParent().getName();
-
```
Comment (dgolovin) [inline src/odo.ts]: Please revert these changes.

File: src/odo.ts [hunk 8]
```diff
@@ -790,13 +796,12 @@ export class OdoImpl implements Odo {
...
     public async _getServices(application: OpenShiftObject): Promise<OpenShiftObject[]> {
         const appName: string = application.getName();
-        const projName: string = application.getParent().getName();
         let services: OpenShiftObject[] = [];
         try {
-            const result: cliInstance.CliExitData = await this.execute(Command.listServiceInstances(projName, appName));
-            services = result.stdout.trim().split('\n')
-                .filter((value) => value !== '')
-                .map((value) => new OpenShiftObjectImpl(application, value, ContextType.SERVICE, true, OdoImpl.instance, TreeItemCollapsibleState.None));
+            const context = OdoImpl.data.getStringByContext(appName);
```
Comment (dgolovin) [inline src/odo.ts]: Remove this line.

File: src/odo.ts [hunk 9]
```diff
@@ -1042,7 +1047,7 @@ export class OdoImpl implements Odo {
         OdoImpl.data.clearTreeData();
     }
 
-    addWorkspaceComponent(folder: WorkspaceFolder, component: OpenShiftObject) {
+    addWorkspaceComponent(folder: WorkspaceFolder, component: OpenShiftObject): void {
```
Comment (dgolovin) [inline src/odo.ts]: Move to different PR, see comment above.

File: src/odo.ts [hunk 10]
```diff
@@ -755,10 +764,7 @@ export class OdoImpl implements Odo {
     }
 
     public async _getStorageNames(component: OpenShiftObject): Promise<OpenShiftObject[]> {
-        const app = component.getParent();
-        const appName = app.getName();
-        const projName = app.getParent().getName();
-        const result: cliInstance.CliExitData = await this.execute(Command.listStorageNames(projName, appName, component.getName()), component.contextPath ? component.contextPath.fsPath : Platform.getUserHomePath());
+        const result: cliInstance.CliExitData = await this.execute(Command.listStorageNames(), component.contextPath ? component.contextPath.fsPath : Platform.getUserHomePath());
```
Comment (dgolovin) [inline src/odo.ts]: Remove from this PR.
